types/ember-truth-helpers/base.d.ts

```diff
@@ -0,0 +1,36 @@
+import Helper from "@glint/environment-ember-loose/ember-component/helper";
+
+interface ComparisonHelperSignature {
+  PositionalArgs: [any, any, { forceNumber?: boolean }?];
+  Return: boolean;
+}
+
+export declare class ComparisonHelper extends Helper<ComparisonHelper> {}
+
+interface EqualityHelperSignature {
+  PositionalArgs: [any, any];
+  Return: boolean;
+}
+
+export declare class EqualityHelper extends Helper<EqualityHelperSignature> {}
+
+// NOTE: These types are imperfect and could definitely use improvement.
+// For instance, the limit at 5 is arbitrary. Also, the `and` and `not` helpers
+// have special handling for empty arrays and objects with an `isTruthy`
+// method. Finally, the actual returned value is the first non-truthy or the _last_
+// item.
+
+interface AndHelperSignature<
+  A,
+  B = never,
+  C = never,
+  D = never,
+  E = never
+> {
+  PositionalArgs: [A, B?, C?, D?, E?];
```

Comment:
For what its worth, I _think_ that something like ```interface AndHelperSignature<A, T extends unknown[]> {
    PositionalArgs: [A, ...T];
    Return: A | NonNullable<T>;
}

export declare class AndHelper<A, T extends unknown[]> extends Helper<AndHelperSignature<A, T>> {}
``` might work to allow us any amount of items

Reply:
This makes me think that having some tests wouldn't be a bad idea :)

Reply:
It turns out it's actually quite complex. I messed around with it for way too long, but I think I got about as good as we can get.